app/forms/UnderpaymentReasonAmendmentFormProvider.scala

```diff
@@ -84,6 +85,26 @@ class UnderpaymentReasonAmendmentFormProvider extends Mappings {
     )
   }
 
+  private def unitsFormMapping: Form[UnderpaymentReasonValue] = {
```

Comment:
This is the same as weightFormMapping so I think the 2 could be combined to be something more generic such as decimalFormMapping? The number of decimal places, the maximum amounts would need to be parameters but I think it should be possible.
You would need to look at how you do the error messages though as they are different between the weight and the units boxes.